Shared/Samples/Search for web map/SearchForWebMapView.Views.swift

```diff
@@ -0,0 +1,109 @@
+    struct SafeMapView: View {
+        /// The map shown in the map view.
+        let map: Map
+        
+        /// A Boolean value indicating whether the map is being loaded.
+        @State private var mapIsLoading = false
+        
+        /// The error shown in the error alert.
+        @State private var error: Error?
+        
+        var body: some View {
+            ZStack {
+                MapView(map: map)
+                    .onLayerViewStateChanged { _, state in
+                        // Show an alert for an error loading any of the layers.
+                        guard let error = state.error else { return }
+                        self.error = error
+                    }
+                    .task {
+                        mapIsLoading = true
+                        defer { mapIsLoading = false }
+                        
+                        // Show an alert for an error loading the map.
+                        do {
+                            try await map.load()
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                
+                if mapIsLoading {
+                    ProgressView()
+                }
+            }
+            .errorAlert(presentingError: $error)
+        }
+    }
+    
+    /// A view that shows a given portal item's info in a row.
+    struct PortalItemRowView: View {
+        /// The portal item to display in the row.
+        let item: PortalItem
+        
+        var body: some View {
+            VStack {
+                HStack {
+                    AsyncImage(url: item.thumbnail?.url) { image in
+                        image
+                            .resizable()
+                    } placeholder: {
+                        Color(.lightGray)
+                    }
+                    .frame(width: 110, height: 75)
+                    .border(.primary)
+                    .padding([.leading, .top], 10)
+                    
+                    Text(item.title)
+                    
+                    Spacer()
+                }
+                
+                HStack {
+                    if let modificationDate = item.modificationDate {
+                        Text(modificationDate, format: Date.FormatStyle(date: .abbreviated, time: .omitted))
+                            .foregroundColor(Color(.systemGray5))
+                    } else {
+                        Text("Date: Unknown")
+                            .foregroundColor(Color(.systemGray5))
+                    }
+                    
+                    Divider()
+                        .overlay(.black)
+                    
+                    Text(item.owner)
+                        .foregroundColor(.teal)
+                    
+                    Spacer()
+                }
+                .font(.footnote)
+                .padding(10)
+                .background(Color(.darkGray))
```

Comment:
The number of differing grays and the teal seem visually busy to me. I tweaked some things around and came up with this, take it or leave it.

```suggestion
                
                Divider()
                    .overlay(.black)
                
                HStack {
                    if let modificationDate = item.modificationDate {
                        Text(modificationDate, format: Date.FormatStyle(date: .abbreviated, time: .omitted))
                            .foregroundColor(Color(.systemGray5))
                    } else {
                        Text("Date: Unknown")
                            .foregroundColor(Color(.systemGray5))
                    }
                    
                    Divider()
                        .overlay(.black)
                    
                    Text(item.owner)
                        .foregroundColor(.accentColor)
                    
                    Spacer()
                }
                .font(.footnote)
                .padding()
```
As is:
<img width="50%" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/16397058/283a18a7-36b3-454b-9fa7-2fbb5863432f">

Suggestion:
<img width="50%" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/16397058/7b2db81b-d8c6-405f-8f42-bbfb0e1da6d3">

The padding in the suggestion could probably be reduced.